kotlin/android/app/src/main/java/dev/firezone/android/features/session/backend/SessionCallbackImpl.kt

```diff
@@ -1,16 +1,24 @@
 package dev.firezone.android.features.session.backend
 
 import android.util.Log
+import dev.firezone.connlib.session.SessionCallback
 
-public object SessionCallbacks {
-    public fun onUpdateResources(resources: String): Boolean {
+class SessionCallbackImpl: SessionCallback {
+
+    override fun onConnect(status: String): Boolean {
+        Log.d("Connlib", "status: $: $status")
+
+        return true
+    }
+
+    override fun onUpdateResources(resources: String): Boolean {
         // TODO: Call into client app to update resources list and routing table
         Log.d("Connlib", "onUpdateResources: $resources")
 
         return true
     }
 
-    public fun onSetTunnelAddresses(addresses: String): Boolean {
+    override fun onSetTunnelAddresses(addresses: String): Boolean {
```

Comment:
`onDisconnect` was added to communicate to the application that the tunnel has been disconnected. See firezone/product#568 and @francesca64 's PR: https://github.com/firezone/firezone/pull/1744/files#diff-02b08c9aa9850edd7d05a5ad8cb81e43b8ab429fc3104d702e8505caab66c355R238

```suggestion
    override fun onDisconnect(): Boolean {
```